code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -770,19 +790,636 @@ nobiliumsuppression = INFINITY
+/datum/gas_reaction/halon_formation
+	priority = 15
+	name = "Halon formation"
+	id = "halon_formation"
+
+/datum/gas_reaction/halon_formation/init_reqs()
+	min_requirements = list(
+		/datum/gas/bz = MINIMUM_MOLE_COUNT,
+		/datum/gas/tritium = MINIMUM_MOLE_COUNT,
+		"TEMP" = 150,
+		"MAX_TEMP" = 250
+	)
+
+/datum/gas_reaction/halon_formation/react(datum/gas_mixture/air, datum/holder)
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/old_heat_capacity = air.heat_capacity()
+	var/heat_efficency = min(temperature * 0.01, cached_gases[/datum/gas/tritium][MOLES], cached_gases[/datum/gas/bz][MOLES])
+	var/energy_used = heat_efficency * 100
+	ASSERT_GAS(/datum/gas/halon, air)
+	if ((cached_gases[/datum/gas/tritium][MOLES] - heat_efficency * 4 < 0 ) || (cached_gases[/datum/gas/bz][MOLES] - heat_efficency * 0.25 < 0)) //Shouldn't produce gas from nothing.
+		return NO_REACTION
+	cached_gases[/datum/gas/tritium][MOLES] -= heat_efficency * 4
+	cached_gases[/datum/gas/bz][MOLES] -= heat_efficency * 0.25
+	cached_gases[/datum/gas/halon][MOLES] += heat_efficency * 4.25
+
+	if(energy_used)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = max(((temperature * old_heat_capacity + energy_used) / new_heat_capacity), TCMB)
+	return REACTING
+
+/datum/gas_reaction/hexane_formation
+	priority = 16
+	name = "Hexane formation"
+	id = "hexane_formation"
+
+/datum/gas_reaction/hexane_formation/init_reqs()
+	min_requirements = list(
+		/datum/gas/bz = MINIMUM_MOLE_COUNT,
+		/datum/gas/hydrogen = MINIMUM_MOLE_COUNT,
+		"TEMP" = 150,
+		"MAX_TEMP" = 250
+	)
+
+/datum/gas_reaction/hexane_formation/react(datum/gas_mixture/air, datum/holder)
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/old_heat_capacity = air.heat_capacity()
+	var/heat_efficency = min(temperature * 0.01, cached_gases[/datum/gas/hydrogen][MOLES], cached_gases[/datum/gas/bz][MOLES])
+	var/energy_used = heat_efficency * 100
+	ASSERT_GAS(/datum/gas/hexane, air)
+	if ((cached_gases[/datum/gas/hydrogen][MOLES] - heat_efficency * 5 < 0 ) || (cached_gases[/datum/gas/bz][MOLES] - heat_efficency * 0.25 < 0)) //Shouldn't produce gas from nothing.
+		return NO_REACTION
+	cached_gases[/datum/gas/hydrogen][MOLES] -= heat_efficency * 5
+	cached_gases[/datum/gas/bz][MOLES] -= heat_efficency * 0.25
+	cached_gases[/datum/gas/hexane][MOLES] += heat_efficency * 5.25
+
+	if(energy_used)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = max(((temperature * old_heat_capacity + energy_used) / new_heat_capacity), TCMB)
+	return REACTING
+
+/datum/gas_reaction/healium_formation
+	priority = 12
+	name = "Healium formation"
+	id = "healium_formation"
+
+/datum/gas_reaction/healium_formation/init_reqs()
+	min_requirements = list(
+		/datum/gas/bz = MINIMUM_MOLE_COUNT,
+		/datum/gas/freon = MINIMUM_MOLE_COUNT,
+		"TEMP" = 50,
+		"MAX_TEMP" = 100
```

Comment:
Ditto

Reply:
i'll change this one to be the "easy" one that can be made at T0C but it has very high heat production that makes it hard to produce near max temp (i was thinking 300 K)